moved/src/move_execution/eth_token.rs

```diff
@@ -45,6 +45,15 @@ pub trait BaseTokenAccounts {
         gas_meter: &mut G,
     ) -> Result<(), crate::Error>;
 
+    fn refund_l2_cost<G: GasMeter>(
```

Comment:
Should we just name this method to just `refund` 🤔?
Here's my thought, from the `BaseTokenAccounts` perspective it would just handle gas costs and refunds. However the layer above will determine the business logic of charging or refunding for L1 or L2. This might be wrong, but it's a thought, since we also change the charging to generic `charge_gas_cost`.

Reply:
Sounds like it would be a clean solution indeed, but my only concern would be if we ever need refunds on L1? AFAICT at least for deposited transactions this explicitly never happens.